builder/frameworks/cmsis.py

```diff
@@ -0,0 +1,521 @@
+  'samg54n19': 'samg54',
+  'samha1g15a': 'samha1a',
+  'samha1g14a': 'samha1a',
+  'samha1g16a': 'samha1a',
+  'samha1e16ab': 'samha1ab',
+  'samha1g15ab': 'samha1ab',
+  'samha1e14ab': 'samha1ab',
+  'samha1g14ab': 'samha1ab',
+  'samha1g16ab': 'samha1ab',
+  'samha1e15ab': 'samha1ab',
+  'saml10e15a': 'saml10',
+  'saml10d16a': 'saml10',
+  'saml10e14a': 'saml10',
+  'saml10d15a': 'saml10',
+  'saml10d14a': 'saml10',
+  'saml10e16a': 'saml10',
+  'saml11e15a': 'saml11',
+  'saml11e14a': 'saml11',
+  'saml11d16a': 'saml11',
+  'saml11d15a': 'saml11',
+  'saml11e16a': 'saml11',
+  'saml11d14a': 'saml11',
+  'saml21e18a': 'saml21a1',
+  'saml21j18a': 'saml21a1',
+  'saml21g18a': 'saml21a1',
+  'saml21j16b': 'saml21b',
+  'saml21j17bu': 'saml21b',
+  'saml21e17b': 'saml21b',
+  'saml21g18b': 'saml21b',
+  'saml21j17b': 'saml21b',
+  'saml21e16b': 'saml21b',
+  'saml21j18bu': 'saml21b',
+  'saml21e18b': 'saml21b',
+  'saml21e15b': 'saml21b',
+  'saml21g16b': 'saml21b',
+  'saml21g17b': 'saml21b',
+  'saml21j18b': 'saml21b',
+  'saml22n18a': 'saml22',
+  'saml22j17a': 'saml22',
+  'saml22g18a': 'saml22',
+  'saml22j16a': 'saml22',
+  'saml22j18a': 'saml22',
+  'saml22g17a': 'saml22',
+  'saml22n17a': 'saml22',
+  'saml22g16a': 'saml22',
+  'saml22n16a': 'saml22',
+  'samr21e17a': 'samr21',
+  'samr21e16a': 'samr21',
+  'samr21g18a': 'samr21',
+  'samr21g16a': 'samr21',
+  'samr21e18a': 'samr21',
+  'samr21g17a': 'samr21',
+  'samr21e19a': 'samr21',
+  'samr30e18a': 'samr30',
+  'samr30g18a': 'samr30',
+  'samr34j18b': 'samr34',
+  'samr34j16b': 'samr34',
+  'samr34j17b': 'samr34',
+  'samr35j18b': 'samr35',
+  'samr35j16b': 'samr35',
+  'samr35j17b': 'samr35',
+  'sams70n20b': 'sams70b',
+  'sams70q20b': 'sams70b',
+  'sams70q19b': 'sams70b',
+  'sams70n19b': 'sams70b',
+  'sams70n21b': 'sams70b',
+  'sams70q21b': 'sams70b',
+  'sams70j20b': 'sams70b',
+  'sams70j19b': 'sams70b',
+  'sams70j21b': 'sams70b',
+  'sams70n21': 'sams70a',
+  'sams70n20': 'sams70a',
+  'sams70n19': 'sams70a',
+  'sams70j20': 'sams70a',
+  'sams70j21': 'sams70a',
+  'sams70q20': 'sams70a',
+  'sams70q21': 'sams70a',
+  'sams70q19': 'sams70a',
+  'sams70j19': 'sams70a',
+  'samv70q19b': 'samv70b',
+  'samv70n19b': 'samv70b',
+  'samv70n20b': 'samv70b',
+  'samv70q20b': 'samv70b',
+  'samv70j19b': 'samv70b',
+  'samv70j20b': 'samv70b',
+  'samv70j19': 'samv70',
+  'samv70q19': 'samv70',
+  'samv70q20': 'samv70',
+  'samv70j20': 'samv70',
+  'samv70n19': 'samv70',
+  'samv70n20': 'samv70',
+  'samv71n19': 'samv71',
+  'samv71n20': 'samv71',
+  'samv71n21': 'samv71',
+  'samv71q19': 'samv71',
+  'samv71j19': 'samv71',
+  'samv71j21': 'samv71',
+  'samv71j20': 'samv71',
+  'samv71q21': 'samv71',
+  'samv71q20': 'samv71',
+  'samv71n19b': 'samv71b',
+  'samv71q19b': 'samv71b',
+  'samv71q21b': 'samv71b',
+  'samv71n21b': 'samv71b',
+  'samv71q20b': 'samv71b',
+  'samv71n20b': 'samv71b',
+  'samv71j19b': 'samv71b',
+  'samv71j21b': 'samv71b',
+  'samv71j20b': 'samv71b',
+}
```

Comment:
Probably we can make it tidier as there is some patterns between mcu and variant name and then refactor this map so it will handle only exceptional cases.

Reply:
Yeah, I considered doing something algorithmic, but there are enough exceptions that the list will probably still be long. I'll take a stab at reducing the patterns. One alternative would be to have a mapping of regexp to processor family.

Reply:
I changed it to a map of mcu_family -> mcu-match-regexp. I confirmed that it returns the same results as explicit map.